FreeRTOS_Sockets.c

```diff
@@ -94,6 +94,32 @@
 #define sock80_PERCENT     80U  /**< 80% of the defined limit. */
 #define sock100_PERCENT    100U /**< 100% of the defined limit. */
 
+/**
+ * @brief Check whether a given socket is valid or not. Validity is defined
+ *        as the socket not being NULL and not being Invalid.
+ * @param[in] xSocket: The socket to be checked.
+ * @return pdTRUE if the socket is valid, else pdFALSE.
+ *
+ */
+portINLINE BaseType_t xSocketValid( Socket_t xSocket )
+{
```

Comment:
Here the same remark as in FreeRTOS_IP.c : would you mind to move it downward?

Reply:
I forgot to mention: `FreeRTOS_round_down()` is not used in the entire project, so I think it can be removed (from IP.c and IP.h).

What about replacing `portINLINE` with an explicit `extern`?
~~~c
	portINLINE ipDECL_CAST_CONST_PTR_FUNC_FOR_TYPE( IPHeader_t );
	extern ipDECL_CAST_CONST_PTR_FUNC_FOR_TYPE( IPHeader_t );
~~~

It makes clear that it has become an external (global) function, and also it looks nicer than this:
~~~c
	ipDECL_CAST_CONST_PTR_FUNC_FOR_TYPE( IPHeader_t );
~~~
Syntactically the above is very strange, it looks like a function `ipDECL_CAST_CONST_PTR_FUNC_FOR_TYPE` without a return type.

One more thing: within the FreeRTOS libraries, it is custom to finish each function with a special line:
~~~c
/*-----------------------------------------------------------*/
~~~
containing 59 dashes.

Reply:
Yes! that is a good suggestion, I added that.